events/bingo/serverScripts/bingoWheel/bingoWheelServer.js

```diff
@@ -376,36 +413,145 @@
+                    lineHeight: 1.58
+                });
+                currentTimeBetweenEdits *= TIME_BETWEEN_EDITS_STEP_MULTIPLIER;
+                bingoWheelEditTimeout = Script.setTimeout(_this.editBingoWheel, currentTimeBetweenEdits);
+            } else {
+                var bingoCall = possibleBingoCalls[Math.floor(Math.random() * possibleBingoCalls.length)];
+                Entities.editEntity(BINGO_TEXT_ENTITY_ID, {
+                    text: bingoCall
+                });
+
+                // Remove the BINGO letter from the call
+                bingoCall = bingoCall.substring(2, bingoCall.length);
+                _this.addCalledNumber(parseInt(bingoCall));
+
+                bingoWheelEditTimeout = false;
+                wheelSpinning = false;
+
+                playSound(BLIP_SOUND, 0.5, GAME_AUDIO_POSITION);
+            }
+        },
+
+        // Starts the Bingo wheel if the user who clicked the wheel is allowed to spin it.
+        // Fills in `possibleBingoCalls` letter/number array based on already-called numbers.
+        spinBingoWheel: function(thisID, params) {
+            var spinnerUsername = params[0];
+
+            if (USERS_ALLOWED_TO_SPIN_WHEEL.indexOf(spinnerUsername) > -1) {
+                if (!gameReady) {
+                    console.log("User attempted to spin wheel, but game is not ready!");
+                    return;
+                }
+
+                if (wheelSpinning) {
+                    return;
+                }
+
+                wheelSpinning = true;
+
+                var i = 1;
+                var currentGeneratedBingoCall;
+                possibleBingoCalls = [];
+
+                while (i < 16) {
+                    currentGeneratedBingoCall = "B " + String(i);
+                    if (calledNumbers.indexOf(i) === -1) {
+                        possibleBingoCalls.push(currentGeneratedBingoCall);
+                    }
+                    i++;
+                }
+                while (i < 31) {
+                    currentGeneratedBingoCall = "I " + String(i);
+                    if (calledNumbers.indexOf(i) === -1) {
+                        possibleBingoCalls.push(currentGeneratedBingoCall);
+                    }
+                    i++;
+                }
+                while (i < 46) {
+                    currentGeneratedBingoCall = "N " + String(i);
+                    if (calledNumbers.indexOf(i) === -1) {
+                        possibleBingoCalls.push(currentGeneratedBingoCall);
+                    }
+                    i++;
+                }
+                while (i < 61) {
+                    currentGeneratedBingoCall = "G " + String(i);
+                    if (calledNumbers.indexOf(i) === -1) {
+                        possibleBingoCalls.push(currentGeneratedBingoCall);
+                    }
+                    i++;
+                }
+                while (i < 76) {
+                    currentGeneratedBingoCall = "O " + String(i);
```

Comment:
I don't think you need the String(i).  I think JS will coerce that to string automatically.

Reply:
i think you are right, but i think this is one case where it's better to be explicit.